### PR TITLE
Added more splitting options

### PR DESCRIPTION
**Pull request by DarkXoa**
Added menus for Specific Missile and Specific Super Missile splits. Locations added as follows:
Alpha Missiles
Charge Missiles
Hi-Jump Missiles
Wave Missiles
Warehouse Missiles
Main Street Missiles
Wrecked Ship Main Shaft Missiles
Early Supers
Phantoon Supers
Crab Supers
Aqueduct Supers
Spore Spawn Supers

Also put a note in the tooltip for Spore Spawn Supers about SSRTA split differences.

### CODE REVIEW

**Comment by DarkXoa** [comment]
I don't really know any of the warps used with spacetime beam, but I did a bit of playing around with it and found that it did split twice with all of my splits, as well as the specific etank splits, enabled, so it will probably be necessary to add room checking to each one. Sorry, chabo. =/

EDIT: I'm not sure where in the check to put the room check, but I'll start adding the relevant room IDs for each of these when I get the chance if someone else hasn't started doing it by then.

File: LiveSplit.SuperMetroid.asl [hunk 1]
```diff
@@ -21,10 +21,38 @@ startup
 	settings.SetToolTip("firstMissile", "Split on the first Missile pickup");
 	settings.Add("allMissiles", false, "All Missiles", "ammoPickups");
 	settings.SetToolTip("allMissiles", "Split on each Missile upgrade");
+	settings.Add("specificMissiles", false, "Specific Missile Packs", "ammoPickups");
+	settings.SetToolTip("ammoPickups", "Split on specific Missile Pack locations");
+	settings.Add("alphaMissiles", false, "Alpha Missile Pack", "specificMissiles");
+	settings.SetToolTip("alphaMissiles", "Split on picking up the Alpha Missile Pack");
+	settings.Add("chargeMissiles", false, "Charge Missile Pack", "specificMissiles");
+	settings.SetToolTip("chargeMissiles", "Split on picking up the Charge Missile Pack");
+	settings.Add("hjMissiles", false, "High-Jump Missile Pack", "specificMissiles");
```
Comment (UNHchabo) [inline LiveSplit.SuperMetroid.asl]: The game spells it as Hi-Jump Boots. Also, I've been avoiding abbreviations in the setting names.

File: LiveSplit.SuperMetroid.asl [hunk 2]
```diff
@@ -21,10 +21,38 @@ startup
...
+	settings.SetToolTip("wsMainShaftMissiles", "Split on picking up the Wrecked Ship Main Shaft Missile Pack");
 	settings.Add("firstSuper", false, "First Supers", "ammoPickups");
 	settings.SetToolTip("firstSuper", "Split on the first Super Missile pickup");
 	settings.Add("allSupers", false, "All Super Missiles", "ammoPickups");
 	settings.SetToolTip("allSupers", "Split on each Super Missile upgrade");
+	settings.Add("specificSupers", false, "Specific Super Missile Packs", "ammoPickups");
+	settings.SetToolTip("ammoPickups", "Split on specific Super Missile Pack locations");
+	settings.Add("earlySupers", false, "Early Supers Missile Pack", "specificSupers");
+	settings.SetToolTip("earlySupers", "Split on picking up the Early Supers Missile Pack");
+	settings.Add("phantoonSupers", false, "Phantoon Super Missile Pack", "specificSupers");
```
Comment (UNHchabo) [inline LiveSplit.SuperMetroid.asl]: Is it common to call them "Phantoon Supers"? I've usually heard Left Side Wrecked Ship.

File: LiveSplit.SuperMetroid.asl [hunk 3]
```diff
@@ -320,6 +348,42 @@ startup
 		{ "chargeBeam",		0x10}
 	};
 
+	vars.missileWatchKeys = new Dictionary<string, string> {
+		{ "alphaMissiles", "brinstarItems4" },
```
Comment (UNHchabo) [inline LiveSplit.SuperMetroid.asl]: I assume these hashes are ordered by how they're usually picked up in 100%?

I think having them grouped by area, then sorted by bitflag order would be neater.

File: LiveSplit.SuperMetroid.asl (hunk 2 above)
Comment (DarkXoa) [reply LiveSplit.SuperMetroid.asl]: Could just be me, but I feel like it's fairly common in any%. Probably should specify for the sake of clarity across categories, though. I'll get all these fixed up and resubmit.

File: LiveSplit.SuperMetroid.asl [hunk 4]
```diff
@@ -863,6 +863,382 @@ gameTime
 	return TimeSpan.FromSeconds(current.totalTime);
 }
 
+isLoading
```
Comment (UNHchabo) [inline LiveSplit.SuperMetroid.asl:866]: Looks like this section got copy-pasted several times at the end of the file.

Comment (DarkXoa) [reply LiveSplit.SuperMetroid.asl:866]: No idea how this happened. I guessing it's something to do with notepad++ on my tablet. Won't use that for edits anymore...

**Pull request merged**